src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace System.Buffers
+{
+    internal sealed class DefaultArrayPool<T> : ArrayPool<T>
+    {
+        private const int MinimiumArraySize = 16;
+        private ArrayPoolBucket<T>[] _buckets;
+
+        internal DefaultArrayPool(int maxLength, int arraysPerBucket)
+        {
+            if (maxLength <= 0)
+                throw new ArgumentOutOfRangeException("maxLength");
+            if (arraysPerBucket <= 0)
+                throw new ArgumentOutOfRangeException("arraysPerBucket");
```

Comment:
Would it make more sense to just make the names the same? Then if we change the default implementation to support more (or more restrictive) preconditions, the Create method wouldn't necessarily have to change to support those too?